src/vs/platform/windows/electron-main/windowImpl.ts

```diff
@@ -958,6 +990,47 @@ export class CodeWindow extends BaseWindow implements ICodeWindow {
 				electron.app.setProxy({ proxyRules, proxyBypassRules, pacScript: '' });
 			}
 		}
+
+		// JS callstack period and frequency
+		if (!e || e.affectsConfiguration('window.unresponsive')) {
+			const unresponsiveWindowSettings = this.configurationService.getValue<IUnresponsiveWindowSettings>('window.unresponsive');
+			// Settings cannot affect current unresponsive window,
+			// so avoid rescheduling here.
+			this._jsCallStackCollector.dispose();
+			this._jsCallStackCollector.delay = unresponsiveWindowSettings.sampleInterval;
+			this._jsCallStackCollectorStopScheduler.cancel();
+			this._jsCallStackCollectorStopScheduler.delay = unresponsiveWindowSettings.samplePeriod;
+		}
+	}
+
+	private async startCollectingJScallStacks(): Promise<void> {
+		if (!this._jsCallStackCollector.isTriggered()) {
+			const stack = await this._win.webContents.mainFrame.collectJavaScriptCallStack();
+			// Increment the count for this stack trace
+			if (stack) {
+				const count = this._jsCallStackMap.get(stack) || 0;
+				this._jsCallStackMap.set(stack, count + 1);
+			}
+			this._jsCallStackCollector.trigger(() => this.startCollectingJScallStacks());
+		}
+	}
+
+	private async stopCollectingJScallStacks(): Promise<void> {
+		this._jsCallStackCollectorStopScheduler.cancel();
+		this._jsCallStackCollector.dispose();
```

Comment:
Why dispose it? What if the window is unresponsive again?

Reply:
Dispose contrary to its name, only seems to cancel the timeout https://github.com/microsoft/vscode/blob/ea9b53fbcbcfd76f57d3f9414fb50ed2d05d3215/src/vs/base/common/async.ts#L429. New unresponsive event will retrigger.

Reply:
will make it dispose => cancel to be explicit.